src/EditorFeatures/CSharpTest/UseNullPropagation/UseNullPropagationTests.cs

```diff
@@ -104,6 +104,54 @@ void M(object o)
         }
 
         [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseNullPropagation)]
+        public async Task TestWithNullableType()
+        {
+            await TestInRegularAndScriptAsync(
+@"
+class C
+{
+    public int? f;
+    void M(C c)
+    {
+        int? x = [||]c != null ? c.f : null;
+    }
+}",
+@"
+class C
+{
+    public int? f;
+    void M(C c)
+    {
+        int? x = c?.f;
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseNullPropagation)]
+        public async Task TestWithNullableTypeAndObjectCast()
+        {
+            await TestInRegularAndScriptAsync(
+@"
+class C
+{
+    public int? f;
+    void M(C c)
+    {
+        int? x = (object)[||]c != null ? c.f : null;
```

Comment:
this is a really cool case.  if you're feeling up to it, you could also check for "obj is null".

Reply:
I looked into it, but for some reason I feel that would not be common (`x is null : null ? x.y`) and the implementation is a bit hairy. Filled follow-up issue.